src/librustc/infer/error_reporting/anon_anon_conflict.rs

```diff
@@ -46,9 +46,9 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
         };
 
         // Determine whether the sub and sup consist of both anonymous (elided) regions.
-        let anon_reg_sup = or_false!(self.is_suitable_anonymous_region(sup));
+        let anon_reg_sup = or_false!(self.is_suitable_region(sup));
```

Comment:
Maybe we should rename this variable, if it's not always anonymous? (For that matter, this file and fn?)

Reply:
How about both_params_anon_or_named_conflict.rs ?

Reply:
hmm, maybe something like `different_lifetimes.rs`? (since this generates the "types declared with different lifetimes..." message)